### PR TITLE
Added external field loading from file to fields.py

### PR DESCRIPTION
**Pull request by camille12225**

### CODE REVIEW

File: PICMI_Python/fields.py [hunk 1]
```diff
@@ -1124,3 +1124,29 @@ def add_refined_region(self, level, lo, hi, refinement_factor=[2,2,2]):
...
+    load_B: bool, default=True
+        If False, do not load magnetic field
+
+    load_E: bool, default=True
+        If False, do not load electric field
+    '''
+    def __init__(self, read_fields_from_path,load_B=True, load_E=True, **kw) :
+        self.load_B = load_B
+        self.load_E = load_E
+        self.read_fields_from_path = read_fields_from_path
```
Comment (dpgrote) [inline PICMI_Python/fields.py:1145]: This routine should call "self.handle_init(kw)". See the other classes defined here as examples.

File: PICMI_Python/fields.py [hunk 2]
```diff
@@ -1124,3 +1124,29 @@ def add_refined_region(self, level, lo, hi, refinement_factor=[2,2,2]):
...
+
+    def initialize_inputs(self):
```
Comment (dpgrote) [inline PICMI_Python/fields.py:1147]: This routine would only be in the WarpX implementation and should not be here.

**Comment by dpgrote** [comment]
Thanks for adding this! I have some comments.

File: PICMI_Python/applied_fields.py [hunk 3]
```diff
@@ -167,3 +167,24 @@ def __init__(self, x_front_location=None, y_front_location=None, z_front_locatio
 
         self.handle_init(kw)
 
+        
+class PICMI_LoadGriddedField(_ClassWithInit):
+    ''' "The data read in is used to initialize the E and B fields on the grid at the start of the simulation.
```
Comment (dpgrote) [inline PICMI_Python/applied_fields.py:172]: This looks good, thanks!
Though sorry for my pickiness, for consistency with the other classes, can you use double quotes here? Also, remove the extra space and double quote at the start of the comment.

File: PICMI_Python/fields.py [hunk 4]
```diff
@@ -1124,3 +1124,4 @@ def add_refined_region(self, level, lo, hi, refinement_factor=[2,2,2]):
             Defaulting to [2,2,2] (relative to next lower level)
         """
         self.refined_regions.append([level, lo, hi, refinement_factor])
+
```
Comment (dpgrote) [inline PICMI_Python/fields.py:1127]: Undo this change that added a newline here.

File: PICMI_Python/applied_fields.py (hunk 3 above)
Comment (camille12225) [reply PICMI_Python/applied_fields.py]: No worries, thanks for being thorough! Fixed

File: PICMI_Python/applied_fields.py [hunk 5]
```diff
@@ -167,3 +167,26 @@ def __init__(self, x_front_location=None, y_front_location=None, z_front_locatio
...
+    read_fields_from_path: string
+        Path to file with field data
+
+    load_B: bool, default=True
+        If False, do not load magnetic field
+
+    load_E: bool, default=True
+        If False, do not load electric field
+    """
+    def __init__(self, read_fields_from_path,load_B=True, load_E=True, **kw) :
```
Comment (dpgrote) [inline PICMI_Python/applied_fields.py:187]:
```suggestion
    def __init__(self, read_fields_from_path, load_B=True, load_E=True, **kw):
```

**Review by dpgrote — APPROVED** [review]
Looks good! Thanks!
